tools/killswitch/src/settings.rs

```diff
@@ -0,0 +1,107 @@
+use crate::errors::Error;
+use nomad_xyz_configuration::{agent::SignerConf, ChainConf, NomadConfig, TxSubmitterConf};
+use std::{collections::HashMap, env, result::Result};
+
+/// KillSwitch `Settings` contains all available configuration for all networks present
+#[derive(Debug)]
+pub(crate) struct Settings {
+    /// NomadConfig
+    pub config: NomadConfig,
+    /// RPC endpoint configs
+    pub rpcs: HashMap<String, ChainConf>,
+    /// Transaction submission configs
+    pub tx_submitters: HashMap<String, TxSubmitterConf>,
+    /// Attestation signer configs
+    pub attestation_signers: HashMap<String, SignerConf>,
+}
+
+impl Settings {
+    /// Build new `Settings` from env and config file
+    pub(crate) async fn new() -> Result<Self, Error> {
+        // Get config
+        let config = if let Ok(config_url) = env::var("CONFIG_URL") {
+            NomadConfig::fetch(&config_url)
+                .await
+                .map_err(|_| Error::BadConfigVar(config_url.clone()))
+        } else if let Ok(config_path) = env::var("CONFIG_PATH") {
+            NomadConfig::from_file(&config_path)
+                .map_err(|_| Error::BadConfigVar(config_path.clone()))
+        } else {
+            Err(Error::NoConfigVar)
+        };
+        let config = config?;
+
+        // Load secrets manually instead of using `AgentSecrets::from_env` so we can load them
+        // best effort instead of bailing on first error
+        let networks = config.networks.clone();
+
+        let rpcs = networks
+            .iter()
+            .filter_map(|n| ChainConf::from_env(&n.to_uppercase()).map(|conf| (n.clone(), conf)))
+            .collect::<HashMap<_, _>>();
+
+        // Load submitter configs for all networks explicitly using the form `<NETWORK>_TXSIGNER_*`
+        let tx_submitters = networks
+            .iter()
+            .filter_map(|n| {
+                TxSubmitterConf::from_env(&n.to_uppercase()).map(|conf| (n.clone(), conf))
+            })
+            .collect::<HashMap<_, _>>();
```

Comment:
Point of clarification: so here we get whatever conf info exists in env and put it in `Settings`. If it turns out we needed a network's details that is not in the settings, we bubble up the error _later_ in the killswitch invocation?

Reply:
So we have 3 places for returning errors:

1. We've hit a complete blocker (e.g., no config file) and return that immediately and exit
2. We fail to init some contracts, which we report when the init step is complete 
3. Final report after tx's have failed / succeeded / timed out 

Every non-blocking error gets collected and reported at step 2.

Only kills that *could* succeed continue and get reported at 3.